stix_shifter_modules/guardium/stix_transmission/api_client.py

```diff
@@ -1,250 +1,73 @@
-                    host_port + '/'  # + endpoint, is set by 'add_endpoint_to_url_header'
-                host_port = proxy.get('x_forward_proxy')
-                if proxy.get('x_forward_proxy_auth', None) is not None:
-                    self.headers['x-forward-auth'] = proxy.get(
-                        'x_forward_proxy_auth')
-                self.headers['user-agent'] = 'UDS'
-                url_modifier_function = self.add_endpoint_to_url_header
-
-        self.client = RestApiClient(host_port,
-                                    None,
-                                    self.headers,
-                                    url_modifier_function,
-                                    cert_verify=connection.get(
-                                        'selfSignedCert', True),
-                                    sni=connection.get('sni', None)
-                                    )
-        self.search_timeout = connection['options'].get('timeout')
-
-    def add_endpoint_to_url_header(self, url, endpoint, headers):
-        # this function is called from 'call_api' with proxy forwarding,
-        # it concatenates the endpoint to the header containing the url.
-        headers['x-forward-url'] += endpoint
-        # url is returned since it points to the proxy for initial call
-        return url
-
-    def add_header_elements(self, key, value):
-        self.headers[key] = value
-        return
-
-    def ping_box(self):
-        # Subroto -- Guardium does not have ping facility
-        # We test if we can get the access token if we can then success = true
-        #
+         # Uncomment when implementing data source API client.
+        # auth = configuration.get('auth')
+        # headers = dict()
+        # headers['X-Auth-Token'] = auth.get('token')
+        # self.client = RestApiClient(connection.get('host'),
+        #                             connection.get('port'),
+        #                             connection.get('cert', None),
+        #                             headers,
+        #                             cert_verify=connection.get('cert_verify', 'True')
+        #                             )
+
+        # Placeholder client to allow dummy transmission calls.
+        # Remove when implementing data source API client.
+        url = "https://"+connection["host"]+":"+connection["port"]
+        self.client = GuardApiClient(connection["options"]["client_id"], url, connection["options"]["client_secret"], \
+                                            configuration["auth"]["username"], configuration["auth"]["password"])
+       
+
+    def ping_data_source(self):
+        # Pings the data source
+        return {"code": 200, "success": True}
+
+
+    def get_search_status(self, search_id):
+        # Check the current status of the search
+        return {"code": 200, "status": "COMPLETED"}
+    
+    def get_status(self, search_id):
+            # Subroto we do not need to send anything to Guardium
```

Comment:
not sure what this `get_status()` function is for. from the implementation it's just querying the api for access token. status needs to return the status of the search/query. I think it is mixed up between ping and search status. 
Does Guardium API returns any search or query status? As a synchronous connector I guess it doesn't. But you should confirm.

Reply:
It is synchronous connector